libpromises/expand.c

```diff
@@ -641,12 +641,73 @@ void BundleResolvePromiseType(EvalContext *ctx, const Bundle *bundle, const char
     }
 }
 
+static int PointerCmp(const void *a, const void *b, ARG_UNUSED void *user_data)
+{
+    if (a < b)
+    {
+        return -1;
+    }
+    else if (a == b)
+    {
+        return 0;
+    }
+    else
+    {
+        return 1;
+    }
+}
+
 void BundleResolve(EvalContext *ctx, const Bundle *bundle)
 {
     Log(LOG_LEVEL_DEBUG,
         "Resolving classes and variables in 'bundle %s %s'",
         bundle->type, bundle->name);
 
+    /* first check if some variables were injected remotely into this bundle and
+     * remove them (CFE-1915) */
+    const Seq *remote_var_promises = EvalContextGetRemoteVarPromises(ctx, bundle->name);
+    if ((remote_var_promises != NULL) && SeqLength(remote_var_promises) > 0)
```

Comment:
Could be nice to put all of this remote var related logic in a separate function.